x/community/keeper/incentives.go

```diff
@@ -1,19 +1,81 @@
 package keeper
 
 import (
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/x/community/types"
 )
 
+// CanPayoutStakingRewards return true if payout staking rewards can be paid out
+func (k Keeper) CanPayoutStakingRewards(ctx sdk.Context) bool {
+	return k.hasExecutedDisableInflationUpgrade(ctx)
+}
+
+// PayoutStakingRewards pays out staking rewards from the community account
+// to the fee collector account for distribution to stakers.
+func (k Keeper) PayoutStakingRewards(ctx sdk.Context) error {
+	params, found := k.GetParams(ctx)
+	if !found {
+		return errorsmod.Wrapf(types.ErrStakingRewardsPayout, "params not found")
+	}
+
+	stakingRewards, err := k.calculateStakingRewards(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleAccountName, k.feeCollectorName, sdk.NewCoins(stakingRewards)); err != nil {
+		return err
+	}
+
+	if stakingRewards.Amount.IsInt64() {
+		defer telemetry.ModuleSetGauge(types.ModuleName, float32(stakingRewards.Amount.Int64()), "payout_staking_rewards")
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypePayoutRewards,
+			sdk.NewAttribute(types.AttributeKeyRewardsPerSecond, params.StakingRewardsPerSecond.String()),
+			sdk.NewAttribute(sdk.AttributeKeyAmount, stakingRewards.Amount.String()),
+		),
+	)
+
+	return nil
+}
+
+func (k Keeper) calculateStakingRewards(ctx sdk.Context) (sdk.Coin, error) {
+	previousBlockTime, found := k.GetPreviousBlockTime(ctx)
+	if !found {
+		return sdk.Coin{}, errorsmod.Wrapf(types.ErrStakingRewardsPayout, "missing previous block time")
+	}
+
+	params, found := k.GetParams(ctx)
+	if !found {
+		return sdk.Coin{}, errorsmod.Wrapf(types.ErrStakingRewardsPayout, "x/community params not found")
+	}
+
+	blockTime := ctx.BlockTime()
+	secondsElapsed := sdkmath.NewInt(blockTime.Unix() - previousBlockTime.Unix())
```

Comment:
Should we use a more precise value for the secondsElapsed? `sdk.Dec` with blocktime ms instead of seconds, then truncate reward coins decimals after multiplying